### ISSUE TITLE
BufferOverrun Analysis Error

### ISSUE THREAD

**Issue by medduk9871** [facebook/infer#584]
Hello!

I tried the Inferbo(Buffer Overrun) with some of open sources.
There's no issue with the test cases in the source code to analyze bufferoverrun.
But, I got the errors like below when analyzing the opensource whose loc is about 362,000.

[Error in analysis phase]
`ONDEMAND EXCEPTION dev_urandom (Failure
  "conversion from int64 to int failed: 9223372036854775807 is out of range")

BACK TRACE
Raised at file "IR/Exceptions.ml", line 308, characters 14-15
Called from file "backend/reporting.ml", line 41, characters 45-79
Called from file "backend/ondemand.ml", line 205, characters 10-112
Called from file "checkers/summary.ml", line 49, characters 11-93
Called from file "bufferoverrun/bufferOverrunChecker.ml", line 249, characters 19-58
Called from file "checkers/AbstractInterpreter.ml", line 76, characters 19-72
Called from file "list.ml", line 84, characters 24-34
Called from file "src/core_list0.ml", line 408, characters 22-47
Called from file "checkers/AbstractInterpreter.ml", line 134, characters 31-86
Called from file "bufferoverrun/bufferOverrunChecker.ml", line 422, characters 18-64
Called from file "checkers/AbstractInterpreter.ml", line 164, characters 12-71
Called from file "checkers/AbstractInterpreter.ml", line 171, characters 13-45
Called from file "backend/ondemand.ml", line 174, characters 4-36

ONDEMAND EXCEPTION random_bytes (Failure
  "conversion from int64 to int failed: 9223372036854775807 is out of range")
`

In other hand. there are no issues analyzing with other checkers like infer, checkers and so on.

Is it the bufferoverrun issue or my environment issue?

**Comment by jvillard** [comment]
Hi @medduk9871, this is an internal infer error. Does the analysis carry through after that? If so you don't need to worry about it.

We can look into fixing it in infer though, thanks for the report.

**Comment by vavrusa** [comment]
It doesn't generate the report fwiw:

```
make: *** [cl231] Error 2
FFFFFFF.F..make: Target `default' not remade because of errors.
Uncaught exception:
  
  (Unix.Exit_or_signal (Exit_non_zero 2))

Raised at file "src/error.ml", line 7, characters 20-30
Called from file "backend/infer.ml", line 340, characters 25-43
Called from file "backend/infer.ml", line 456, characters 2-21
```

**Comment by jvillard** [comment]
cc @KihongHeo

**Comment by KihongHeo** [comment]
Hi @vavrusa ,
Did you run the buffer overrun checker? What kind of options did you use? 
And please provide more detailed information about the error such as your environment, options, or minimal crashing examples.

**Comment by vavrusa** [comment]
Hi, I ran the buffer overrun checker in OS X.

```bash
$ brew install infer
$ infer --version
Infer version v0.10.0
Copyright 2009 - present Facebook. All Rights Reserved.
$ uname -rsv
Darwin 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64
```

I'm having a hard time to create a minimal example though, the constructions I thought could be a problem seem to be okay when I create an example. Here's what I did:

```bash
git clone https://github.com/CZ-NIC/knot.git && cd knot
brew install automake autoconf libtool pkg-config gnutls jansson nettle userspace-rcu
autoreconf -fvi
./configure
infer -a bufferoverrun -- make
```

**Comment by KihongHeo** [comment]
Thanks @vavrusa. Now I figured out the problem. I will submit the patch today. 

Thanks for your help.

**Comment by jvillard** [comment]
This should be fixed in the latest release thanks to 89a28d4
